float32[] state
---
float64 x1
float64 x2